components/spicedb/codegen/codegen.go

```diff
@@ -107,13 +110,90 @@ func GenerateDefinition(schema *compiler.CompiledSchema) string {
 		}
 	}
 
-	return `/**
- * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
- * Licensed under the GNU Affero General Public License (AGPL).
- * See License.AGPL.txt in the project root for license information.
- */
+	return `
+		/**
+		* Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+		* Licensed under the GNU Affero General Public License (AGPL).
+		* See License.AGPL.txt in the project root for license information.
+		*/
 
-export const InstallationID = "1";
-` +
-		resource + ";\n\n" + relation + ";\n\n" + permission + ";\n\n" + other
+		import { v1 } from "@authzed/authzed-node";
+
+		export const InstallationID = "1";
+
+		` + resource + `;
+
+		` + relation + `;
+
+		` + permission + `;
+
+		` + other + `
+
+		export const rel = {
+			` + fluentApi + `
+		};
+		`
+}
+
+func generateFluentAPI(def *corev1.NamespaceDefinition) string {
+	ifElse := func(cond bool, ifTrue string, ifFalse string) string {
+		if cond {
+			return ifTrue
+		}
+		return ifFalse
+	}
+
+	relations := ""
+	for _, rel := range def.Relation {
+		if namespace.GetRelationKind(rel) != implv1.RelationMetadata_RELATION {
+			continue
+		}
+
+		objects := ``
+		for _, t := range rel.TypeInformation.AllowedDirectRelations {
+			objects += `
+				` + ifElse(t.Namespace == "installation",
+				"get "+t.Namespace+"()",
+				t.Namespace+"(objectId: string)") + ` {
+				return {
+					...result2,
+					subject: {
+						object: {
+							objectType: "` + t.Namespace + `",
+							objectId: ` + ifElse(t.Namespace == "installation", `"1"`, "objectId") + `,
```

Comment:
```suggestion
							objectId: ` + ifElse(t.Namespace == "installation", `InstallationID`, "objectId") + `,
```

Reply:
This is generated code. I will check if we still need the other constants in a follow up PR and also add a big comment to the top of the file.

Reply:
Oh, I thought this is the generator?

Reply:
ah, yes you are right, it is the generator

Reply:
:relieved: